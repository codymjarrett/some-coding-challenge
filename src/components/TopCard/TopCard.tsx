import React from 'react'

export interface TopCardInputProps {
	movie: any
}

const TopCard: React.SFC<TopCardInputProps> = ({ movie }) => (
	<div className="topCard">
		{movie.fun_facts ? (
			<div className="topCard__content">
				<p className="detail-highlight">Fun Fact</p>
				<p className="topCard__detail"> {movie.fun_facts}</p>
			</div>
		) : null}
		{movie.production_company ? (
			<div className="topCard__content">
				<p className="detail-highlight">Production Company</p>{' '}
				<p className="topCard__detail">{movie.production_company}</p>
			</div>
		) : null}
		{movie.distributor ? (
			<div className="topCard__content">
				<p className="detail-highlight">Distributor</p>{' '}
				<p className="topCard__detail">{movie.distributor}</p>
			</div>
		) : null}
	</div>
)

export default TopCard
