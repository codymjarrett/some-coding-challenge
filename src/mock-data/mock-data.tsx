export const mockData = () => {
	return {
		title: "Alexander's Ragtime Band",
		release_year: '1938',
		locations: [
			'Cliff House (1090 Point Lobos Avenue)',
			'The Walden House, Buena Vista Park',
		],
		fun_facts:
			'In 1887, the Cliff House was severely damaged when the schooner Parallel, abandoned and loaded with dynamite, ran aground on the rocks below.',
		production_company: 'Twentieth Century Fox Film Corporation',
		distributor: 'Twentieth Century Fox Film Corporation',
		director: 'Henry King',
		writer: 'Kathryn Scola',
		actor_1: 'Tyrone Power',
		actor_2: 'Alice Faye',
		actor_3: 'Don Ameche',
		actors: ['Tyrone Power', 'Alice Faye', 'Don Ameche'],
	}
}
