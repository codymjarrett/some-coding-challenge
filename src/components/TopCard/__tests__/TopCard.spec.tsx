import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { mockData } from '../../../mock-data/mock-data'

import TopCard from '../TopCard'

afterEach(cleanup)

it('<TopCard/>', () => {
	const { container, debug } = render(<TopCard movie={mockData} />)
	expect(container.firstChild).toMatchSnapshot()
	debug()
})
