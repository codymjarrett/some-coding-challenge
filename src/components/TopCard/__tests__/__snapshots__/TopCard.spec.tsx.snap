// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopCard/> 1`] = `
<div
  class="topCard"
/>
`;
